base/sysinfo.jl

```diff
@@ -411,7 +412,18 @@ including e.g. a WebAssembly JavaScript embedding in a web browser.
 """
 isjsvm(os::Symbol) = (os === :Emscripten)
 
-for f in (:isunix, :islinux, :isbsd, :isapple, :iswindows, :isfreebsd, :isopenbsd, :isnetbsd, :isdragonfly, :isjsvm)
+"""
+    Sys.detectwsl([os])
+
+Runtime Predicate for testing if Julia is running inside WSL.
```

Comment:
How about this? (The 1.11 would have to be changed if this were to be backported to 1.10)
```suggestion
Runtime predicate for testing if Julia is running inside
Windows Subsystem for Linux (WSL).

!!! note
    Unlike `Sys.iswindows`, `Sys.islinux` etc., this is a runtime tests, and thus
    cannot meaningfully be used in `@static if ` constructs.

!!! compat "Julia 1.11"
    This function requires at least Julia 1.11.
```